lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -123,7 +123,7 @@ impl Clone for LockedSegment {
     }
 
     fn clone_from(&mut self, source: &Self) {
-        *self = source.clone();
+        self.clone_from(source);
     }
 }
```

Comment:
I also caught this one locally.

I think we should just remove this function, it is provided by the trait: https://doc.rust-lang.org/std/clone/trait.Clone.html#method.clone_from